src/Cech_complex/utilities/delaunay_cech_persistence.cpp

```diff
@@ -0,0 +1,159 @@
+                                            false, true);
+  Gudhi::cech_complex::assign_MEB_filtration(Kernel(), stree, point_cloud, exact_version);
+  stree.prune_above_filtration(max_radius);
+  return stree;
+}
+
+int main(int argc, char* argv[]) {
+  std::string off_file_points;
+  std::string pers_diag_file;
+  bool exact_version = false;
+  bool fast_version = false;
+  Filtration_value max_radius;
+  int p;
+  Filtration_value min_persistence;
+
+  program_options(argc, argv, off_file_points, exact_version, fast_version, pers_diag_file, max_radius, p,
+                  min_persistence);
+
+  if ((exact_version) && (fast_version)) {
+    std::cerr << "You cannot set the exact and the fast version." << std::endl;
+    exit(-1);
+  }
+
+  Simplex_tree stree;
+  if (fast_version) {
+    // WARNING : CGAL::Epick_d is fast but not safe (unlike CGAL::Epeck_d)
+    // (i.e. when the points are on a grid)
+    using Fast_kernel = CGAL::Epick_d<CGAL::Dynamic_dimension_tag>;
+    stree = create_simplex_tree<Fast_kernel>(off_file_points, exact_version, max_radius);
+  } else {
+    std::clog << "exact_version = " << exact_version << "\n";
+    using Kernel = CGAL::Epeck_d<CGAL::Dynamic_dimension_tag>;
+    stree = create_simplex_tree<Kernel>(off_file_points, exact_version, max_radius);
+  }
+
+  std::clog << "The complex contains " << stree.num_simplices() << " simplices \n";
+  std::clog << "   and has dimension " << stree.dimension() << " \n";
+
+  // Sort the simplices in the order of the filtration
+  stree.initialize_filtration();
+
+  // Compute the persistence diagram of the complex
+  Persistent_cohomology pcoh(stree);
+  // initializes the coefficient field for homology
+  pcoh.init_coefficients(p);
+
+  pcoh.compute_persistent_cohomology(min_persistence);
+
+  // Output the persitence diagram in pers_diag_file
+  if (pers_diag_file.empty()) {
+    pcoh.output_diagram();
+  } else {
+    std::ofstream out(pers_diag_file);
+    pcoh.output_diagram(out);
+    out.close();
+  }
+
+  return 0;
+}
+
+void program_options(int argc, char* argv[], std::string& off_file_points, bool& exact, bool& fast,
+                     std::string& pers_diag_file, Filtration_value& max_radius, int& p,
+                     Filtration_value& min_persistence) {
+  namespace po = boost::program_options;
+  po::options_description hidden("Hidden options");
+  hidden.add_options()("input-file", po::value<std::string>(&off_file_points),
+                       "Name of an OFF file containing a point set.\n");
+
+  po::options_description visible("Allowed options", 100);
+  visible.add_options()("help,h", "produce help message")(
+      "exact,e", po::bool_switch(&exact),
+      "To activate exact version of Delaunay-Cech complex (default is false, not available if fast is set)")(
+      "fast,f", po::bool_switch(&fast),
+      "To activate fast version of Delaunay-Cech complex (default is false, not available if exact is set)")(
```

Comment:
(I realize that this is probably copied from another utility)
It isn't super clear what exact/fast mean precisely. We should say somewhere:
1. fast: right combinatorics, values can be arbitrarily bad
2. default, safe: values can have a relative error at most XXX
3. exact: true values rounded to double.

Reply:
To be discussed on #1029